src/main/java/ch/naviqore/raptor/router/DateTimeUtils.java

```diff
@@ -0,0 +1,48 @@
+package ch.naviqore.raptor.router;
+
+import ch.naviqore.raptor.TimeType;
+
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+class DateTimeUtils {
+
+    static LocalDate getReferenceDate(Map<?, LocalDateTime> sourceStops, TimeType timeType) {
+        if (timeType == TimeType.DEPARTURE) {
+            // get minimum departure time
+            return sourceStops.values()
+                    .stream()
+                    .min(Comparator.naturalOrder())
+                    .map(LocalDateTime::toLocalDate)
+                    .orElseThrow();
+        } else {
+            // get maximum arrival time
+            return sourceStops.values()
+                    .stream()
+                    .max(Comparator.naturalOrder())
+                    .map(LocalDateTime::toLocalDate)
+                    .orElseThrow();
+        }
+    }
+
+    static Map<String, Integer> mapLocalDateTimeToTimestamp(Map<String, LocalDateTime> sourceStops,
+                                                            LocalDate referenceDate) {
+        return sourceStops.entrySet()
+                .stream()
+                .map(e -> Map.entry(e.getKey(), convertToTimestamp(e.getValue(), referenceDate)))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    static LocalDateTime convertToLocalDateTime(int timestamp, LocalDate referenceDate) {
+        return referenceDate.atStartOfDay().plusSeconds(timestamp);
+    }
+
+    static int convertToTimestamp(LocalDateTime localDateTime, LocalDate referenceDate) {
```

Comment:
Would this not be convertToSecondsOfDay? Since the timestamp with origin 1970 is removed?